src/scenes/Activities/scenes/EventHistory/index.js

```diff
@@ -0,0 +1,249 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import Paper from 'material-ui/Paper';
+import Grid from 'material-ui/Grid';
+import Typography from 'material-ui/Typography';
+import Table, { TableBody, TableCell, TableHead, TableRow, TableSortLabel } from 'material-ui/Table';
+import Tooltip from 'material-ui/Tooltip';
+import { withStyles } from 'material-ui/styles';
+import { FormattedMessage, injectIntl, intlShape, defineMessages } from 'react-intl';
+import moment from 'moment';
+
+import styles from './styles';
+import Config from '../../../../config/app';
+import graphqlActions from '../../../../redux/Graphql/actions';
+import { getShortLocalDateTimeString } from '../../../../helpers/utility';
+
+class EventHistory extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      order: 'asc',
+      orderBy: 'time',
+      transactions: [],
+    };
+
+    this.getTableHeader = this.getTableHeader.bind(this);
+    this.getTableRows = this.getTableRows.bind(this);
+    this.createSortHandler = this.createSortHandler.bind(this);
+    this.handleSorting = this.handleSorting.bind(this);
+    this.executeTxsRequest = this.executeTxsRequest.bind(this);
+  }
+
+  componentWillMount() {
+    this.executeTxsRequest();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {
+      getTransactionsReturn,
+      syncBlockTime,
+    } = nextProps;
+
+    // Update page on new block
+    if (syncBlockTime !== this.props.syncBlockTime) {
+      this.executeTxsRequest();
+    }
+
+    this.setState({ transactions: getTransactionsReturn });
+  }
+
+  render() {
+    const { transactions } = this.state;
+
+    return (
+      <Grid container spacing={0}>
+        {
+          transactions.length ?
+            (<Table>
+              {this.getTableHeader()}
+              {this.getTableRows(this.state.transactions)}
+            </Table>) :
+            (<Typography variant="body1">
+              You do not have any transactions right now.
+            </Typography>)
+        }
+      </Grid>
+    );
+  }
+
+  executeTxsRequest() {
+    this.props.getTransactions([], undefined);
```

Comment:
you need to add filters for this one. need to add all of the types except for `Transfer` type